nixos/modules/services/networking/netbird.md

```diff
@@ -2,55 +2,113 @@
 
 ## Quickstart {#module-services-netbird-quickstart}
 
-The absolute minimal configuration for the netbird daemon looks like this:
+The absolute minimal configuration for the Netbird client daemon looks like this:
 
 ```nix
 services.netbird.enable = true;
 ```
 
-This will set up a netbird service listening on the port `51820` associated to the
-`wt0` interface.
-
-It is strictly equivalent to setting:
+Which is equivalent to:
 
 ```nix
-services.netbird.tunnels.wt0.stateDir = "netbird";
+services.netbird.clients.wt0 = {
+  port = 51820;
+  name = "netbird";
+  interface = "wt0";
+};
 ```
 
-The `enable` option is mainly kept for backward compatibility, as defining netbird
-tunnels through the `tunnels` option is more expressive.
+This will set up a `netbird.service` listening on the port `51820` associated to the
+`wt0` interface. There will also be `netbird-wt0` binary installed in addition to `netbird`.
+
+see [the interpolation section](#module-services-netbird-interpolation) for more details.
 
 ## Multiple connections setup {#module-services-netbird-multiple-connections}
 
-Using the `services.netbird.tunnels` option, it is also possible to define more than
+Using the `services.netbird.clients` option, it is possible to define more than
 one netbird service running at the same time.
 
-The following configuration will start a netbird daemon using the interface `wt1` and
-the port 51830. Its configuration file will then be located at `/var/lib/netbird-wt1/config.json`.
+You must at least define a `port` for the service to listen on, the rest is optional:
 
 ```nix
-services.netbird.tunnels = {
-  wt1 = {
-    port = 51830;
-  };
-};
+services.netbird.clients.wt1.port = 51830;
+services.netbird.clients.wt2.port = 51831;
 ```
 
-To interact with it, you will need to specify the correct daemon address:
+see [the interpolation section](#module-services-netbird-interpolation) for more details.
+
+## Exposing services internally on the Netbird network {#module-services-netbird-firewall}
+
+You can easily expose services exclusively to Netbird network by combining
+[`networking.firewall.interfaces`](#opt-networking.firewall.interfaces) rules with [`interface`](#opt-services.netbird.clients._name_.interface) names:
 
-```bash
-netbird --daemon-addr unix:///var/run/netbird-wt1/sock ...
+```nix
+{
+  services.netbird.clients.priv.port = 51819;
+  services.netbird.clients.work.port = 51818;
+  networking.firewall.interfaces = {
+    "${config.services.netbird.clients.priv.interface}" = {
+      allowedUDPPorts = [ 1234 ];
+    };
+    "${config.services.netbird.clients.work.interface}" = {
+      allowedTCPPorts = [ 8080 ];
+    };
+  };
+}
 ```
 
-The address will by default be `unix:///var/run/netbird-<name>`.
+## Configuration interpolation {#module-services-netbird-interpolation}
```

Comment:
The term interpolation seems weird to me in this context, maybe use `interface-specific` ?

Reply:
yeah, I'm not sure what to call it either, how about *Configuration overview*?